package/MDAnalysis/lib/util.py

```diff
@@ -391,6 +391,8 @@ def anyopen(datasource, mode='rt', reset=True):
         else:
             stream = None
             filename = datasource
+            if filename.endswith('pdb') or filename.endswith('xyz'):
+                return pickle_open(filename, mode=mode)
```

Comment:
pickle_open was added here (instead of inside PDBReader/XYZReader) because `anyopen` also support a stream as input.